src/wms/src/main/java/org/geoserver/wms/map/GetMapKvpRequestReader.java

```diff
@@ -764,6 +796,35 @@ private InputStream getHttpInputStream(URL styleUrl) throws IOException {
         }
     }
 
+    /** Executes the HTTP request with the max request time settings. */
+    private CloseableHttpResponse executeRequest(HttpCacheContext cacheContext, HttpGet httpget)
+            throws IOException, ClientProtocolException {
+        int hardTimeout = 60000;
+        String maxTimeStr =
+                GeoServerExtensions.getProperty(GETMAPKVPREQUESTREADER_MAXREQUEST_TIME_VARIABLE);
+        if (StringUtils.isNotEmpty(maxTimeStr)) {
+            try {
+                hardTimeout = Integer.parseInt(maxTimeStr);
+            } catch (NumberFormatException ex) {
+                LOGGER.log(
+                        Level.FINE,
+                        "Bad number format on mas request time variable, proceeding to use the default.",
+                        ex);
+            }
+        }
+        TimerTask task =
```

Comment:
You should has some comment explaining what's going on here, I guess you use the time to implement the time-out?

Reply:
Could [this](https://stackoverflow.com/questions/3000214/java-http-client-request-with-defined-timeout) be a better option?

Reply:
Yeah, the timeout should be defined at the http client level.

Reply:
- The provided link is about the Connection Timeout.  This timer bound execution is for limiting the max whole-request time, for which there is not an available setting on HttpClient API. This approach follows a recommended strategy for aborting the request with a hard time limit from some technical articles about HttpClient usage.
- Connection and Socket data packet timeout configuration is provided on the ```GETMAPKVPREQUESTREADER_TIMEOUT``` global setting and sets the connection and socket timeout in the HttpClient API since it's available.